### PR TITLE
[DEV-1380] - Create VPC and Security Group for Strapi

### PR DESCRIPTION
**Pull request by maurodandrea**
Create Network VPC and Security Group for Strapi

#### List of Changes
Create Terraform file for networking

#### Motivation and Context
Create Network VPC and Security Group for Strapi

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [X] My change not requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: .infrastructure/14_network.tf [hunk 1]
```diff
@@ -0,0 +1,118 @@
+## Network VPC and Security Group for CMS Strapi
+data "aws_availability_zones" "available" {
+  state = "available"
+}
+
+locals {
+  region   = "eu-south-1"
+  vpc_cidr = "10.0.0.0/16"
+  azs      = slice(data.aws_availability_zones.available.names, 0, 3)
+}
```
Comment (datalek) [inline .infrastructure/14_network.tf:10]: I would remove them, `region` and `vpc_cidr` are not used, `azs` is used only on vpc

**Review by datalek — CHANGES_REQUESTED** [review]

File: .infrastructure/98_variables.tf [hunk 2]
```diff
@@ -60,3 +60,8 @@ variable "log_retention_days" {
   description = "The number of days logs should be retained. Default is 90 days."
   default     = 90
 }
+
+## CMS Strapi App Port
+variable "cms_app_port" {
+  default = 1337
```
[the remark this answers is not in the document]
Comment (maurodandrea) [reply .infrastructure/98_variables.tf]: change made with last commit

File: .infrastructure/14_network.tf (hunk 1 above)
Comment (maurodandrea) [reply .infrastructure/14_network.tf]: change made with last commit

File: .infrastructure/14_network.tf [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    to_port     = 443
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  ingress {
+    protocol    = "tcp"
+    from_port   = var.cms_app_port
+    to_port     = var.cms_app_port
+    cidr_blocks = ["0.0.0.0/0"]
+  }
```
Comment (datalek) [inline .infrastructure/14_network.tf:54]: Why the system needs this ingress config? I would remove it

**Review by datalek — CHANGES_REQUESTED** [review]

File: .infrastructure/14_network.tf (hunk 3 above)
Comment (maurodandrea) [reply .infrastructure/14_network.tf:54]: it should not be needed, as the ALB will only have listeners on http:80 (redirect to https:443) and https:443. I remove this rule

**Review by datalek — APPROVED** [review]

File: .infrastructure/14_network.tf [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  ## https://developer.hashicorp.com/terraform/language/meta-arguments/lifecycle#create_before_destroy
```
Comment (datalek) [inline .infrastructure/14_network.tf:56]:
```suggestion
  # https://registry.terraform.io/providers/hashicorp/aws/5.36.0/docs/resources/security_group#recreating-a-security-group
```

File: .infrastructure/14_network.tf [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+  }
+
+  egress {
+    protocol    = "-1"
+    from_port   = 0
+    to_port     = 0
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  ## https://developer.hashicorp.com/terraform/language/meta-arguments/lifecycle#create_before_destroy
```
Comment (datalek) [inline .infrastructure/14_network.tf:83]:
```suggestion
  # https://registry.terraform.io/providers/hashicorp/aws/5.36.0/docs/resources/security_group#recreating-a-security-group
```

File: .infrastructure/14_network.tf [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
```
Comment (datalek) [inline .infrastructure/14_network.tf:112]:
```suggestion
  # https://registry.terraform.io/providers/hashicorp/aws/5.36.0/docs/resources/security_group#recreating-a-security-group
```

**Review by datalek — CHANGES_REQUESTED** [review]

**Review by datalek — APPROVED** [review]

**Pull request merged**